h2o-algos/src/main/java/water/userapi/Dataset.java

```diff
@@ -0,0 +1,145 @@
+package water.userapi;
+
+import water.Key;
+import water.Scope;
+import water.fvec.*;
+import water.parser.BufferedString;
+import water.parser.ParseDataset;
+import water.rapids.ast.prims.advmath.AstStratifiedSplit;
+import water.udf.specialized.Enums;
+import water.util.FileUtils;
+import water.util.RandomUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static water.rapids.ast.prims.advmath.AstStratifiedSplit.*;
+/**
+ * Simplified Frame wrapper for simple usages
+ * 
+ * Created by vpatryshev on 2/20/17.
+ */
+public class Dataset {
+  Frame frame;
+
+  private Dataset(Frame frame) {
+    this.frame = frame;
+  }
+
+  public static Dataset readFile(String path) {
+    return read(FileUtils.locateFile(path));
+  }
+  
+  public static Dataset read(File file) {
+    try {
+      FileUtils.checkFile(file, file.getCanonicalPath());
+      NFSFileVec nfs = NFSFileVec.make(file);
+      return new Dataset(Scope.track(ParseDataset.parse(Key.make(), nfs._key)));
+    } catch (IOException ioe) {
+      throw new DataException("Could not read " + file, ioe);
+    }
+  }
+  
+  public static Dataset onVecs(Map<String, Vec> map) {
+    return new Dataset(new Frame(
+        map.keySet().toArray(new String[map.size()]),
+        map.values().toArray(new Vec[map.size()])));
+  }
+  
+  public Dataset oneHotEncode(String... ignore) {
+    try {
+      return new Dataset(Enums.oneHotEncoding(frame, ignore));
+    } catch (IOException ioe) {
+      throw new DataException("Failed to do oneHotEncoding", ioe);
+    }
+  }
+  
+  // the new vec is named (hard-coded so far) "test_train_split"
+  Dataset stratifiedSplit(String colName, Double ratio, long seed) {
+    return new Dataset(AstStratifiedSplit.split(frame, frame.vec(colName), ratio, seed, SplittingDom));
+  }
+
+  public TrainAndValid stratifiedSplit(String colName, Double ratio) {
+    Dataset blend = stratifiedSplit(colName, ratio, System.currentTimeMillis());
+    Map<String, Frame> split = blend.splitBy(TestTrainSplitColName, SplittingDom);
+    
+    return new TrainAndValid(
+        new Dataset(split.get(SplittingDom[0])),
+        new Dataset(split.get(SplittingDom[1])));
+  }
+
+  private Frame select(final String what, String colname) {
```

Comment:
When I see a select method on a Frame, what Im usually expecting is to select some set of columns from the Frame. Here it looks like im filtering for rows of the frame where "what" is equal to some value in colname
Take a look at:  https://spark.apache.org/docs/1.6.0/api/java/index.html . There is a select method example. 
I know the method is private, so Im not saying this is wrong, just providing a suggestion for future funtionality